sdk/core/Azure.Core/src/Shared/OperationHelpers.cs

```diff
@@ -53,5 +56,70 @@ public static async ValueTask<Response<TResult>> DefaultWaitForCompletionAsync<T
                 await Task.Delay(pollingInterval, cancellationToken).ConfigureAwait(false);
             }
         }
+
+        public static Response<TResult> DefaultWaitForCompletion<TResult>(this Operation<TResult> operation, CancellationToken cancellationToken)
+            where TResult : notnull
+        {
+            return operation.DefaultWaitForCompletion(DefaultPollingInterval, cancellationToken);
+        }
+
+        public static Response<TResult> DefaultWaitForCompletion<TResult>(
```

Comment:
The `WaitForCompletionAsync` implementation in `OperationInternal` uses the service's `retry-after` header (when available) to set the polling interval. Should we do the same in these `DefaultWaitForCompletion` implementations?

Reply:
Looking at this again, none of the wait methods are ever referenced, so I think they aren't needed anymore.

Reply:
I've refactored the wait logic to OperationHelpers now.